src/impl_float_maths.rs

```diff
@@ -0,0 +1,157 @@
+//! Element-wise methods for ndarray
+
+use num_traits::Float;
+
+use crate::imp_prelude::*;
+
+macro_rules! boolean_op {
+    ($($(#[$meta1:meta])* fn $id1:ident $(#[$meta2:meta])* fn $id2:ident -> $func:ident)+) => {
+        $($(#[$meta1])*
+        pub fn $id1(&self) -> Array<bool, D> {
+            self.mapv(A::$func)
+        }
+        $(#[$meta2])*
+        pub fn $id2(&self) -> bool {
+            self.mapv(A::$func).iter().any(|&b|b)
+        })+
+    };
+}
+
+macro_rules! unary_op {
+    ($($(#[$meta:meta])* fn $id:ident)+) => {
+        $($(#[$meta])*
+        pub fn $id(&self) -> Array<A, D> {
+            self.mapv(A::$id)
+        })+
+    };
+}
+
+macro_rules! binary_op {
+    ($($(#[$meta:meta])* fn $id:ident($ty:ty))+) => {
+        $($(#[$meta])*
+        pub fn $id(&self, rhs: $ty) -> Array<A, D> {
+            self.mapv(|v| A::$id(v, rhs))
+        })+
+    };
+}
+
+/// # Element-wise methods for Float Array
+///
+/// Element-wise math functions for any array type that contains float number.
+impl<A, S, D> ArrayBase<S, D>
+where
+    A: Float,
+    S: RawData<Elem = A> + Data,
+    D: Dimension,
+{
+    boolean_op! {
+        /// If the number is `NaN` (not a number), then `true` is returned for each element.
+        fn is_nan
+        /// Return `true` if any element is `NaN` (not a number).
+        fn is_nan_any -> is_nan
+
+        /// If the number is infinity, then `true` is returned for each element.
+        fn is_infinite
+        /// Return `true` if any element is infinity.
+        fn is_infinite_any -> is_infinite
+    }
+    unary_op! {
+        /// The largest integer less than or equal to each element.
+        fn floor
+
+        /// The smallest integer less than or equal to each element.
+        fn ceil
+
+        /// The nearest integer of each element.
+        fn round
+
+        /// The integer part of each element.
+        fn trunc
+
+        /// The fractional part of each element.
+        fn fract
+
+        /// Absolute of each element.
+        fn abs
+
+        /// Sign number of each element.
+        ///
+        /// + `1.0` for all positive numbers.
+        /// + `-1.0` for all negative numbers.
+        /// + `NaN` for all `NaN` (not a number).
+        fn signum
+
+        /// The reciprocal (inverse) of each element, `1/x`.
+        fn recip
+
+        /// Square root of each element.
+        fn sqrt
+
+        /// `e^x` of each element. (Exponential function)
+        fn exp
+
+        /// `2^x` of each element.
+        fn exp2
+
+        /// Natural logarithm of each element.
+        fn ln
+
+        /// Base 2 logarithm of each element.
+        fn log2
+
+        /// Base 10 logarithm of each element.
+        fn log10
+
+        /// Cubic root of each element.
+        fn cbrt
+
+        /// Sine of each element. (in radians)
```

Comment:
The full stop should be at the end of the sentence here, as it is in `num_traits`